rust/worker/src/execution/operators/fetch_segment.rs

```diff
@@ -0,0 +1,194 @@
+use chroma_blockstore::provider::BlockfileProvider;
+use chroma_error::{ChromaError, ErrorCodes};
+use chroma_index::{hnsw_provider::HnswIndexProvider, IndexConfig, IndexConfigFromSegmentError};
+use chroma_types::{Collection, Segment, SegmentScope, SegmentType};
+use thiserror::Error;
+use tonic::async_trait;
+use tracing::trace;
+use uuid::Uuid;
+
+use crate::{
+    execution::operator::Operator,
+    segment::{
+        distributed_hnsw_segment::{
+            DistributedHNSWSegmentFromSegmentError, DistributedHNSWSegmentReader,
+        },
+        metadata_segment::{MetadataSegmentError, MetadataSegmentReader},
+        record_segment::{RecordSegmentReader, RecordSegmentReaderCreationError},
+    },
+    sysdb::sysdb::{GetCollectionsError, GetSegmentsError, SysDb},
+};
+
+#[derive(Clone, Debug)]
+pub struct FetchSegmentOperator {
+    // Data provider
+    pub(crate) sysdb: Box<SysDb>,
+    pub hnsw: HnswIndexProvider,
+    pub blockfile: BlockfileProvider,
+    // pub knn: Uuid,
+    pub metadata: Uuid,
```

Comment:
how do you decide what goes in here v/s what does in the input?

Reply:
I put everything that can be determined without executing the operators it depends on inside the operator, and anything else will be left as input to the operator. This allows me to directly store query operators inside query orchestrators, which makes it similar to the query plan interface.